packages/react/src/patterns/sub-patterns/TableOfContents/TableOfContents.js

```diff
@@ -33,11 +31,49 @@ const { prefix } = settings;
  */
 const TableOfContents = ({ menuItems, children, menuLabel, theme }) => {
   const [selectedId, setSelectedId] = useState(menuItems[0].id);
-  const [selectedTitle, setSelectedTitle] = useState(menuItems[0].title);
-
+  const [selectedTitle, setSelectedTitle] = useState(menuItems[0].name);
+  const [autoMenuItems, setmenuItems] = useState([]);
   useEffect(() => {
+    /**
+     * Set selected id & title
+     *
+     */
+    if (menuItems) {
```

Comment:
The order of this seems all off. If anything, if `menuItems` doesn't exist, then you should be running `findMenuItems`, correct? It seems like `setSelectedItem` should be run regardless of whether `menuItems` exists or not. The only difference is that if it already is passed in as props, it won't run `findMenuItems`, otherwise it will.